src/Workspaces/CSharp/Portable/Formatting/Rules/SuppressFormattingRule.cs

```diff
@@ -286,6 +289,109 @@ private void AddStatementExceptBlockSuppressOperations(List<SuppressOperation> l
             AddSuppressWrappingIfOnSingleLineOperation(list, firstToken, lastToken);
         }
 
+        private void AddFormatSuppressOperations(List<SuppressOperation> list, SyntaxNode node)
+        {
+            foreach (var child in node.ChildNodesAndTokens())
+            {
+                if (!child.IsToken)
+                {
+                    continue;
+                }
+
+                ProcessTriviaList(list, child.AsToken().LeadingTrivia);
+                ProcessTriviaList(list, child.AsToken().TrailingTrivia);
+            }
+
+            // Local functions
+            static void ProcessTriviaList(List<SuppressOperation> list, SyntaxTriviaList triviaList)
+            {
+                foreach (var trivia in triviaList)
+                {
+                    ProcessTrivia(list, trivia);
+                }
+            }
+
+            static void ProcessTrivia(List<SuppressOperation> list, SyntaxTrivia trivia)
+            {
+                if (!(trivia.HasStructure))
+                {
+                    return;
+                }
+
+                ProcessStructuredTrivia(list, trivia.GetStructure());
+            }
+
+            static void ProcessStructuredTrivia(List<SuppressOperation> list, SyntaxNode structure)
+            {
+                if (!(structure is PragmaWarningDirectiveTriviaSyntax pragmaWarningDirectiveTrivia))
+                {
+                    return;
+                }
+
+                if (!IsFormatDirective(pragmaWarningDirectiveTrivia, SyntaxKind.DisableKeyword))
+                {
+                    return;
+                }
+
+                var startToken = pragmaWarningDirectiveTrivia.GetLastToken(includeZeroWidth: true, includeDirectives: true);
+
+                var endDirective = pragmaWarningDirectiveTrivia.GetNextDirective(trivia => IsFormatDirective(trivia, SyntaxKind.RestoreKeyword));
+                var endToken = endDirective is null
+                    ? structure.SyntaxTree.GetRoot().GetLastToken(includeZeroWidth: true)
+                    : endDirective.GetFirstToken(includeDirectives: true);
+
+                if (startToken.IsKind(SyntaxKind.None) || endToken.IsKind(SyntaxKind.None))
+                {
```

Comment:
should we assert here? this shouldn't happen, right?

Reply:
Correct, changed to an assertion